include/seqan3/core/algorithm/parameter_pack.hpp

```diff
@@ -0,0 +1,87 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2018, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2018, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+#pragma once
+
+#include <utility>
+
+#include <seqan3/std/concepts>
+#include <seqan3/std/type_traits>
+
+namespace seqan3::detail
+{
+
+/*!\brief Applies a function to a range of arguments.
+ * \ingroup algorithm
+ *
+ * \tparam unary_function_t The function type, like function pointers, functors and lambdas.
+ * \tparam ...args_t        The parameter pack of the arguments. (Note that each argument type can be different)
+ *
+ * \param  fn               The function to call on every argument.
+ * \param  args             The arguments as parameter pack.
+ *
+ * \details This function behaves like std::for_each but on parameter packs.
+ * The invocation will be done without any loop.
+ *
+ * \include test/snippet/core/algorithm/for_each_value.cpp
+ */
+template <typename unary_function_t, typename ...args_t>
+//!\cond
+    requires (std::Invocable<unary_function_t, args_t> && ... && true)
+//!\endcond
+constexpr void for_each_value(unary_function_t && fn, args_t && ...args)
+{
+    [[maybe_unused]] int r = (fn(std::forward<args_t>(args)), ..., 0);
+}
+
+/*!\brief Applies a function to a range of types.
+ * \ingroup algorithm
+ *
+ * \tparam ...types         The parameter pack of types.
+ * \tparam unary_function_t The function type, like function pointers, functors and lambdas.
+ *
+ * \param  fn               The function to call on every std::type_identity<type>.
+ *
+ * \details This function behaves like std::for_each but on types. The type will be wrapped into std::type_identity and
+ * passed as argument. The invocation will be done without any loop.
+ *
+ * \include test/snippet/core/algorithm/for_each_type.cpp
+ */
+template <typename ...types, typename unary_function_t>
+//!\cond
+    requires (std::Invocable<unary_function_t, std::type_identity<types>> && ... && true)
```

Comment:
I am not sure if I understand correctly why you have to use type_identity in this context. Can you please elaborate your rationale?

Reply:
I wrap the type within `std::type_identity<>` and call the function. The functions must therefore be callable with `type_identity`